src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_enkf_update.F90

```diff
@@ -2932,6 +2935,114 @@ end subroutine write_smapL4SMaup
 
   ! -----------------------------------------------------------------
 
+!sqz 2020 -----
+  ! **********************************************************
+
+ subroutine get_diagaup (option, N_ens, N_catl, cat_param, cat_progn, diagaup) 
```

Comment:
I'm not sure why this subroutine needs to distinguish between the "forecast" and the "analysis".
In a nutshell, we should only need a wrapper for catch_calc_soilmoist(), catch_calc_tsurf(), and catch_calc_tp().
If we send the updated variables (ie, cat_progn after the increments have been applied) into the subroutine, we should then get the desired *_ana diagnostics back.

Reply:
currently we borrowed different variable names in the "cat_progn" type  to hold "fcst" (5) and "ana" (5),  "forecast" or "analysis" tells the subroutine which variable names ( in the type)  to be put into for locally calculated tsurf etc.  the "type" is the one to communicate between local routine and the assimGC.   
if we skip "fcst" in the export, we do not need "forecast" or "analysis" here.